### ISSUE TITLE
I have connected the MFRC522 correctly but my tag isn't detected

### ISSUE THREAD

**Issue by andreagiov** [omersiar/esp-rfid#447]

**Comment by matjack1** [comment]
hey @andrea006 if you are still interested can you please try using the `dev` branch and build the debug target so that we can see more output on the serial port and I can try helping you troubleshooting. Good luck!

**Comment by mkrasselt1** [comment]
just to make shure you are using the right token - does your anroid phone detect the token when you hold it against the nfc antenna ? because if not, it might be a 125kHZ token and unsuaitable for this project

**Comment by windy54** [comment]
I have got this issue with the dev branch and a debug build.
The hardware is okay because I rolled back to the version we use in our hackspace and that reads cards.

**Comment by windy54** [comment]
I have investigated this with my MFRC522 reader and from what I can see rfidState is never set to card swiped.
mfrc522Read updates the uid and type but does not update rfidState.
So I have updated as shown below.
`#ifndef OFFICIALBOARD
void mfrc522Read()
{
	
	if (!mfrc522.PICC_IsNewCardPresent() || !mfrc522.PICC_ReadCardSerial())
	{
		return;
	}
	
	mfrc522.PICC_HaltA();
	cooldown = millis() + COOLDOWN_MILIS;
#ifdef DEBUG
	Serial.print(F("[ INFO ] nfrc522Read PICC's UID: "));
#endif
	for (int i = 0; i < mfrc522.uid.size; ++i)
	{
		uid += String(mfrc522.uid.uidByte[i], HEX);
	}
#ifdef DEBUG
	Serial.print(uid);
#endif
	MFRC522::PICC_Type piccType = mfrc522.PICC_GetType(mfrc522.uid.sak);
	type = mfrc522.PICC_GetTypeName(piccType);
	rfidState = cardSwiped;
#ifdef DEBUG
	Serial.println(" " + type);
#endif
}`

Adding the line rfidState = cardSwiped afer the card type is read.

I am still investigating as the card was read, but when I updated the card information the ESP re-set.

I will post further updates and then maybe try a pull request if I sort it.

Stevce

**Comment by windy54** [comment]
When I made the above change "Beeper ON" was constantly being output on the serial interface making it difficult to see what was happening.
This was caused by this line in beeperBeep
if (config.beeperpin != 255)

beeperpin was 0 even though I set it to none in the config page.

So my configuration is not being saved correctly .
I overcame it temporarily be editing the config.esp to force beeperpin to 255

When I scan a card in the database the door is opened, when I scan a new cardthe prompt to enter details appears and looks as if it has been saved. If I scan the card again it is treated as a new user.

So I have a problem where configuration and new users are not saved correctly.
Steve
